tensorflow_quantum/python/layers/circuit_executors/input_checks.py

```diff
@@ -0,0 +1,157 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Input checks common to circuit execution layers."""
+import numpy as np
+import sympy
+import tensorflow as tf
+
+import cirq
+from tensorflow_quantum.python import util
+
+
+def expand_circuits(inputs, symbol_names=None, symbol_values=None):
+    """Function for consistently expanding circuit inputs.
+
+    Args:
+        inputs: a single `cirq.Circuit`, a Python `list` or `tuple` of
+            `cirq.Circuit`s, or a pre-converted `tf.Tensor` of
+            `cirq.Circuit`s.
+        symbol_names: a Python `list`, `tuple`, or `numpy.ndarray`
+            of `str` or `sympy.Symbols`, or a `tf.Tensor` of dtype `string`.
+            These are the symbols parameterizing the input circuits.
+        symbol_values: a Python `list`, `tuple`, or `numpy.ndarray` of
+            floating point values, or `tf.Tensor` of dtype `float32`.
+
+    Returns:
+        inputs: `tf.Tensor` of dtype `string` with shape [batch_size]
+            containing the serialized circuits to be executed.
+        symbol_names: `tf.Tensor` of dtype `string` with shape [n_params], which
+            is used to specify the order in which the values in
+            `symbol_values` should be placed inside of the circuits.
+        symbol_values: `tf.Tensor` of dtype `float32` with shape
+            [batch_size, n_params] specifying parameter values to resolve
+            into the circuits following the ordering dictated by `symbol_names`.
+    """
+    # inputs is the circuit(s).
+    symbols_empty = False
+    if symbol_names is None:
+        symbol_names = []
+    if symbol_values is None:
+        symbols_empty = True
+        symbol_values = [[]]
+
+    # Ingest and promote symbol_names.
+    if isinstance(symbol_names, (list, tuple, np.ndarray)):
+        if isinstance(symbol_names, np.ndarray):
+            symbol_names = symbol_names.tolist()
+        if symbol_names and not all(
+            [isinstance(x, (str, sympy.Symbol)) for x in symbol_names]):
+            raise TypeError("Each element in symbol_names"
+                            " must be a string or sympy.Symbol.")
+        symbol_names = [str(s) for s in symbol_names]
+        if not len(symbol_names) == len(list(set(symbol_names))):
+            raise ValueError("All elements of symbol_names must be unique.")
+        symbol_names = tf.convert_to_tensor(symbol_names,
+                                            dtype=tf.dtypes.string)
+    elif tf.is_tensor(symbol_names):
+        if not symbol_names.dtype == tf.dtypes.string:
+            raise TypeError("symbol_names tensor must have dtype string.")
+        if not symbol_names.shape[0] == len(list(set(symbol_names.numpy()))):
```

Comment:
These checks rely on having access to `.numpy()` and `.shape[0]`. In the past these have failed when wrapped in `@tf.function` despite passing bazel tests. These tests are also doing data type checking which is done at the op level so it might be best to just revert this block to how it looked before.